server/item/book_and_quill.go

```diff
@@ -0,0 +1,98 @@
+package item
+
+// BookAndQuill is an item used to write WrittenBook(s).
+type BookAndQuill struct {
+	// Pages represents the pages within the book.
+	Pages []string
+}
+
+// MaxCount always returns 1.
+func (w BookAndQuill) MaxCount() int {
+	return 1
+}
+
+// ValidPage checks to see whether a page exists or not.
+func (w BookAndQuill) ValidPage(page int) bool {
+	return page >= 0 && len(w.Pages) > page
+}
+
+// Page returns a specific page from the book. If the page does not exist
+func (w BookAndQuill) Page(page int) (string, bool) {
+	if page < 0 {
+		panic("negative page number")
+	}
+	if len(w.Pages) <= page {
+		return "", false
+	}
+	return w.Pages[page], true
+}
+
+// SetPage writes a page to the book, if the page doesn't exist it will be created. It will panic if the
+// text is longer then 256 characters. It will return a new book representing this data.
+func (w BookAndQuill) SetPage(page int, text string) BookAndQuill {
+	if page < 0 || page >= 50 {
+		panic("invalid page number")
+	}
+	if len(text) > 256 {
+		panic("text longer then 256 bytes")
+	}
+	if !w.ValidPage(page) {
+		pages := make([]string, page+1)
+		copy(pages, w.Pages)
+		w.Pages = pages
+	}
+	w.Pages[page] = text
+	return w
+}
+
+// SwapPages swaps two different pages, it will panic if the largest of the two numbers doesn't exist. It will
+// return the newly updated pages.
+func (w BookAndQuill) SwapPages(pageOne, pageTwo int) BookAndQuill {
+	if pageOne < 0 || pageTwo < 0 {
+		panic("negative page number")
+	}
+	if !w.ValidPage(max(pageOne, pageTwo)) {
+		panic("invalid page number")
+	}
+	contentOne := w.Pages[pageOne]
+	contentTwo := w.Pages[pageTwo]
+	w.Pages[pageOne] = contentTwo
+	w.Pages[pageTwo] = contentOne
```

Comment:
```suggestion
	temp := w.Pages[pageOne]
	w.Pages[pageOne] = w.Pages[pageTwo] 
	w.Pages[pageTwo] = temp
```

and frankly variable names `a`/ `b` rather than the longer `pageOne`/`pageTwo` are preferred in this specific context with the name & docs already providing enough information